src/metabase/db.clj

```diff
@@ -368,10 +368,12 @@
   ;; Establish our 'default' DB Connection
   (create-connection-pool! (jdbc-details db-details))
 
-  ;; Do any custom code-based migrations now that the db structure is up to date
-  ;; NOTE: we use dynamic resolution to prevent circular dependencies
-  (require 'metabase.db.migrations)
-  ((resolve 'metabase.db.migrations/run-all)))
+  ;; Do any custom code-based migrations now that the db structure is up to date.
+  ;; None of the migrations should be ran when Metabase is launched via `load-from-h2`. That's because they will end up doing things like
+  ;; creating duplicate entries for the "magic" groups and permissions entries. They appropriate things will be copied over from the H2 database.
+  (when-not (= config/launch-command :load-from-h2)
```

Comment:
this makes me feel really itchy.

hardcoding the specific command + a global state